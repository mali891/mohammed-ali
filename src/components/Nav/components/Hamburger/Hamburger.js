import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { styles } from './Hamburger.styles'

const Hamburger = ({ menuExpanded, onClick }) => {
  const dynamicClassNames = classNames([], {
    'c-hamburger': true,
    'c-hamburger--collapse': true,
    'c-hamburger--is-active': menuExpanded
  })

  return (
    <button
      css={styles}
      className={dynamicClassNames}
      type="button"
      aria-label="Menu"
      aria-controls="navigation"
      aria-expanded={menuExpanded}
      onClick={onClick}
    >
      <span className="c-hamburger__box">
        <span className="c-hamburger__inner u-bg--secondary"></span>
      </span>
    </button>
  )
}

Hamburger.propTypes = {
  menuExpanded: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
}

export default Hamburger
