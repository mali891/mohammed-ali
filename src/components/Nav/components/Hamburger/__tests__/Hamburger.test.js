import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Hamburger from '../Hamburger'

describe('Hamburger', () => {
  const mockOnClickHandler = jest.fn()
  const defaultProps = {
    menuExpanded: false,
    onClick: mockOnClickHandler
  }

  it('renders correctly', () => {
    const { asFragment } = render(<Hamburger {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given a menuExpanded prop', () => {
    const { asFragment, container } = render(<Hamburger {...defaultProps} menuExpanded={true} />)

    expect(container.querySelector('.c-hamburger--is-active')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given an onClick prop', () => {
    const { container } = render(<Hamburger {...defaultProps} onClick={mockOnClickHandler} />)
    const hamburger = container.querySelector('.c-hamburger')

    expect(mockOnClickHandler).toHaveBeenCalledTimes(0)
    fireEvent.click(hamburger)
    expect(mockOnClickHandler).toHaveBeenCalledTimes(1)
  })

  // it('renders correctly given a noHover prop', () => {
  //   const { asFragment, container } = render(<Hamburger noHover />)

  //   expect(container.querySelector('.c-Hamburger--no-hover')).toBeInTheDocument()
  //   expect(asFragment()).toMatchSnapshot()
  // })

  // it('renders correctly when given a specific to prop', () => {
  //   const { asFragment, container } = render(<Hamburger type="anchor" to="test.com" />)

  //   expect(container.querySelector('a[href="test.com"]')).toBeInTheDocument()
  //   expect(asFragment()).toMatchSnapshot()
  // })

  // it('renders correctly when given a gatsby type prop', () => {
  //   const { asFragment } = render(<Hamburger type="gatsby" />)

  //   expect(asFragment()).toMatchSnapshot()
  // })

  // it('renders correctly when given an anchor type prop', () => {
  //   const { asFragment, container } = render(<Hamburger type="anchor" />)

  //   expect(container.querySelector('a')).toBeInTheDocument()
  //   expect(asFragment()).toMatchSnapshot()
  // })

  // it('renders correctly when given a button type prop', () => {
  //   const { asFragment, container } = render(<Hamburger type="button" />)

  //   expect(container.querySelector('button')).toBeInTheDocument()
  //   expect(asFragment()).toMatchSnapshot()
  // })

  // it('renders correctly when given an onClick prop', () => {
  //   const mockOnClickHandler = jest.fn()
  //   const { asFragment, container } = render(<Hamburger onClick={mockOnClickHandler} />)
  //   const Hamburger = container.querySelector('.c-Hamburger')

  //   expect(mockOnClickHandler).toHaveBeenCalledTimes(0)
  //   fireEvent.click(Hamburger)
  //   expect(mockOnClickHandler).toHaveBeenCalledTimes(1)
  //   expect(asFragment()).toMatchSnapshot()
  // })
})
