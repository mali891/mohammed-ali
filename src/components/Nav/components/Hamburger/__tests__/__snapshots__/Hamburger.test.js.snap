// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-controls="navigation"
    aria-expanded="false"
    aria-label="Menu"
    class="c-hamburger c-hamburger--collapse"
    css="[object Object]"
    type="button"
  >
    <span
      class="c-hamburger__box"
    >
      <span
        class="c-hamburger__inner u-bg--secondary"
      />
    </span>
  </button>
</DocumentFragment>
`;

exports[`Hamburger renders correctly given a menuExpanded prop 1`] = `
<DocumentFragment>
  <button
    aria-controls="navigation"
    aria-expanded="true"
    aria-label="Menu"
    class="c-hamburger c-hamburger--collapse c-hamburger--is-active"
    css="[object Object]"
    type="button"
  >
    <span
      class="c-hamburger__box"
    >
      <span
        class="c-hamburger__inner u-bg--secondary"
      />
    </span>
  </button>
</DocumentFragment>
`;
